api/renter.go

```diff
@@ -334,6 +334,20 @@ func (api *API) renterDownloadHandler(w http.ResponseWriter, req *http.Request,
 	WriteSuccess(w)
 }
 
+// renterDownloadAsyncHandler handles the API call to download a file asynchronously.
+func (api *API) renterDownloadAsyncHandler(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	destination := req.FormValue("destination")
+	// Check that the destination path is absolute.
+	if !filepath.IsAbs(destination) {
+		WriteError(w, Error{"destination must be an absolute path"}, http.StatusBadRequest)
+		return
+	}
+
+	go api.renter.Download(strings.TrimPrefix(ps.ByName("siapath"), "/"), destination)
```

Comment:
One thing to consider here is error handling. The `DownloadInfo` struct doesn't contain an error field, so there's no way for the caller to tell if an async download failed.

We should add an `Err` field. We may also want to modify the `Download` call itself to be async, and then have the synchronous API call do polling. On the other hand, that would probably cause quite a few tests to fail...